src/test/java/org/example/Operations/QueryOperationsTest.java

```diff
@@ -0,0 +1,140 @@
+        String sql = "select token_id from R_TOKN_MAIN where token_name = 'rodsgroup'";
+
+        // Show that rodsusers are NOT allowed to add specific queries.
+        res = rods.queryOperations().addSpecificQuery(aliceToken, specificQueryName, sql);
+        logger.debug(res.getBody());
+        assertEquals("Adding specific query request failed", 200, res.getHttpStatusCode());
+        // ierror code of -13000: SYS_NO_API_PRIV
+        assertEquals("Adding specific query was supposed to fail", -13000,
+                getIrodsResponseStatusCode(res.getBody()));
+
+        try {
+            // Show that only rodsadmin are allowed to add specific queries.
+            res = rods.queryOperations().addSpecificQuery(rodsToken, specificQueryName, sql);
+            logger.debug(res.getBody());
+            assertEquals("Adding specific query request failed", 200, res.getHttpStatusCode());
+            assertEquals("Adding specific query failed",0,
+                    getIrodsResponseStatusCode(res.getBody()));
+
+            // Use the new specific query.
+            res = rods.queryOperations().executeSpecificQuery(aliceToken, specificQueryName, null);
+            logger.debug(res.getBody());
+            assertEquals("Executing specific query request failed", 200, res.getHttpStatusCode());
+            assertEquals("Executing specific query failed",0,
+                    getIrodsResponseStatusCode(res.getBody()));
+
+            // Show that rodsusers are NOT allowed to remove specific queries.
+            res = rods.queryOperations().removeSpecificQuery(aliceToken, specificQueryName);
+            logger.debug(res.getBody());
+            assertEquals("Removing specific query request failed", 200, res.getHttpStatusCode());
+            // ierror code of -13000: SYS_NO_API_PRIV
+            assertEquals("Removing specific query was supposed to fail", -13000,
+                    getIrodsResponseStatusCode(res.getBody()));
+
+            // Show that only rodsadmins are allowed to remove specific queries.
+            res = rods.queryOperations().removeSpecificQuery(rodsToken, specificQueryName);
+            logger.debug(res.getBody());
+            assertEquals("Removing specific query request failed", 200, res.getHttpStatusCode());
+            // ierror code of -13000: SYS_NO_API_PRIV
+            assertEquals("Removing specific query was failed", 0,
+                    getIrodsResponseStatusCode(res.getBody()));
+        } finally {
+            // Remove specific query
+            rods.queryOperations().removeSpecificQuery(rodsToken, specificQueryName);
+        }
+    }
+
+    private int getIrodsResponseStatusCode(String jsonString) {
+        JsonNode rootNode = assertDoesNotThrow(() -> mapper.readTree(jsonString),
+                "JsonProcessingException was thrown");
+        JsonNode statusCodeNode = rootNode.path("irods_response").path("status_code");
+        return statusCodeNode.asInt();
+    }
```

Comment:
Is this defined in other test modules too? It looks familiar.

Reply:
Yes it's in the other tests as well.

Reply:
See https://github.com/irods/irods_client_http_java/pull/50#discussion_r1700464128.

Reply:
Ok, sounds good. Created #54